apps/devhub/tasks.py

```diff
@@ -180,13 +181,24 @@ def handle_upload_validation_result(results, upload_pk, annotate=True):
     now = datetime.datetime.now()
     now_ts = utc_millesecs_from_epoch(now)
     delta = now_ts - upload_start
+    statsd.timing('devhub.validation_results_processed', delta)
+
+    # Scale the upload / processing time by MB size of the package
+    # so we can normalize large XPIs which naturally take longer to validate.
+    size = storage.size(upload.path)
+    scaled_delta = None
+    if size > 0:
+        size_in_mb = Decimal(size) / Decimal(1024 * 1024)
+        scaled_delta = Decimal(delta) / size_in_mb
+        statsd.timing('devhub.validation_results_processed_per_mb',
+                      scaled_delta)
```

Comment:
I don't think validation is exactly proportionate to size -- that is, a 10MB file won't take *exactly* 10 times longer. But when running our load tests with the larger file, it definitely made validation slower (I'm ignoring upload throughput time). This is probably related simply to the I/O and processing time it takes the validator to scan each file.

That said, it's not a perfect metric but I think it will give us a more realistic view into performance. For example, if some part of the system *unrelated* to the I/O issue I mentioned above is slow then this metric will expose it.

The problem I'm seeing now is that when one person uploads a large XPI, we get alerted that "add-on submission is slow" when really someone just uploaded a big file.